### PR TITLE
Add model layer, move there the the generating points logic.

### PR DESCRIPTION
**Pull request by kirpichonok**
# What

Add class `PointsGeneratorModel` to encapsulate the generating points logic.

# How 

- Add class `PointsGeneratorModel`

- Move generating points logic from the `MainScreenVM` to the `PointsGeneratorModel`.

### CODE REVIEW

File: Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import Foundation
+
+class PointsGeneratorModel {
```
Comment (vkostechko-applace) [inline Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift:3]: final?

File: Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    /// Point to be updated over time.
+    @Published var point: CGPoint?
```
Comment (vkostechko-applace) [inline Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift:5]: private(set)?

File: Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        }
+    }
+
+    // MARK: - Private
+
+    private var counter = 3
+    private var timeInterval = 2.0
+    private var timer: Timer?
+
+    private func reset() {
```
Comment (vkostechko-applace) [inline Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift:33]: please keep file structure 

- properties (public/private)
- init / overrides 
- public methods
- private methods
- protocols

File: Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        timer?.invalidate()
+        counter = 3
```
Comment (vkostechko-applace) [inline Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift:35]: I suggest to move 3 to constants 
BirbankInvest_—_ProcessingScreenViewModelImpl_swift

**Review by vkostechko-applace — CHANGES_REQUESTED** [review]

File: Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift (hunk 3 above)
Comment (kirpichonok) [reply Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift:35]: Fixed.

File: Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift (hunk 1 above)
Comment (kirpichonok) [reply Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift]: Fixed.

File: Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift (hunk 4 above)
Comment (kirpichonok) [reply Auto-Clicker-App/Auto-Clicker-App/MainScreen/PointsGeneratorModel.swift:37]: We're going to have settings for these values, and I use an **instance** of the `PointsGeneratorModel` class. I think it's more appropriate to use instance properties to change the behavior of the specified model instance.
It's good idea to set static property for `maxAmoutOfPointers = 5` or `maxRetryDelay: TimeInterval = 60.0` (according to design model).
WDYT?

**Review by vkostechko-applace — APPROVED** [review]

**Pull request merged**